laser scans not showing when fixed frame is not equal to the laser scans frame id

I see this on Ubuntu 18.04 with the bouncy deb binaries.

Steps to reproduce:

- `ros2 launch dummy_robot_bringup dummy_robot_bringup.launch.py`
- `rviz2`
- Add a display by topic and use `/scan`
- Add the tf display
- Add a map display for `/map`
- Set the fixed frame to `world`

You'll see the map and the tf frames moving, but no laser scan. If you look at the laser display you'll see that the status is `Ok`, but the Points field just says `Showing [0] points from [0] messages` and it is not rendered. I would expect it to be rendered.

If you change the fixed frame to `single_rrbot_hokuyo_link` you can see the laser scans and the status adds the `Transform: Ok` status. However, the map no longer renders and prints to the console `[ERROR] [rviz2]: Error transforming map 'Map' from frame 'world' to 'single_rrbot_hokuyo_link'.` and it also sets a similar message, `Could not transform from [world] to [single_rrbot_hokuyo_link]` in the status part of the map display.

My expectation is that both the laser and map should work in either (or any) fixed frame.

Since #375 has been merged, following the steps to reproduce, I can now see the laser scan in all frames. But, I am unable to see the map when switching to the frames that are not static wrt `world`, ie. `single_rrbot_hokuyo_link`, `single_rrbot_camera_link`, `single_rrbot_link2`, and `single_rrbot_link3`.

I get the error:

```
[ERROR] [rviz2]: Lookup would require extrapolation into the future.  Requested time 1565204390.71718 but the latest data is at time 1565204390.71718, when looking up transform from frame [single_rrbot_hokuyo_link] to frame [world]
```

--- a/rviz_default_plugins/src/rviz_default_plugins/displays/map/map_display.cpp
+++ b/rviz_default_plugins/src/rviz_default_plugins/displays/map/map_display.cpp
@@ -573,10 +573,11 @@ void MapDisplay::transformMap()
 
   Ogre::Vector3 position;
   Ogre::Quaternion orientation;
-  if (!context_->getFrameManager()->transform(frame_, transform_time, current_map_.info.origin,
-    position, orientation) &&
-    !context_->getFrameManager()->transform(frame_, context_->getClock()->now(),
-    current_map_.info.origin, position, orientation))
+  if (!context_->getFrameManager()->transform(
+      frame_, transform_time, current_map_.info.origin, position, orientation) &&
+    !context_->getFrameManager()->transform(
+      frame_, rclcpp::Time(0, 0, context_->getClock()->get_clock_type()),
+      current_map_.info.origin, position, orientation))
   {
     setMissingTransformToFixedFrame(frame_);
     scene_node_->setVisible(false);

